### PR TITLE
Fix broken test suite

### LINKED ISSUE #56: broccoli-persistent-filter upstream dep update broke tests
Found here: https://github.com/kellyselden/broccoli-jscs/issues/27#issuecomment-167599851

> I figure it has to be an npm dependency change. rimraf and glob dependencies for broccoli-persistent-filter have bumped up in the last month (from 2.4.4 -> 2.5.0 and 5.0.15 -> 6.0.2 respectively). No CHANGELOG on either of those modules so that's just a shot in the dark.

If I revert #41, it goes back to working.

cc @rwjblue @ronco

### PR DESCRIPTION
**Pull request by Turbo87**
This PR fixes the broken test suite by replacing the `chdir()` calls.

resolves #56

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -40,9 +40,10 @@ var JSCSFilter = function(inputTree, _options) {
   }
 
   if (this.enabled) {
-    this.rules = config.load(this.configPath || '.jscsrc') || this.config || {};
+    this.rules = config.load(this.configPath || (inputTree + '/.jscsrc')) || this.config || {};
```
Comment (rwjblue) [inline index.js]: `inputTree` is not guaranteed to be  a string.

**Comment by rwjblue** [comment]
Looks good other than that one nit...

File: index.js (hunk 1 above)
Comment (Turbo87) [inline index.js]: argh, that's what I was afraid of. your jshint plugin seems to use `inputPaths[0]` instead, but it seems that that isn't available at this point...

**Comment by Turbo87** [comment]
@rwjblue I've moved the rules loading step into the `build()` method now, roughly similar to how the jshint plugin is doing it. This defers the loading step until `inputPaths` is available so we can use that instead of the `inputTree`.

**Comment by Turbo87** [comment]
@kellyselden could you restart that appveyor build? looks like a timeout, but I don't think it's related to these changes.

**Comment by kellyselden** [comment]
@Turbo87 restarted

**Comment by rwjblue** [comment]
This looks good to me. 

@Turbo87 - Thank you for tackling this!

@kellyselden - Sorry for borking things up in https://github.com/kellyselden/broccoli-jscs/pull/41.

**Comment by Turbo87** [comment]
@rwjblue thanks for the review!

@kellyselden looks like this is ready for merging now :shipit:

**Comment by kellyselden** [comment]
@rwjblue @Turbo87 Thanks guys!

**Comment by ronco** [comment]
Thanks guys!

**Pull request merged**